src/NRediSearch/Aggregation/AggregationBuilder.cs

```diff
@@ -0,0 +1,157 @@
+﻿// .NET port of https://github.com/RedisLabs/JRediSearch/
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using NRediSearch.Aggregation.Reducers;
+
+namespace NRediSearch.Aggregation
+{
+    public class AggregationBuilder
+    {
+        private readonly List<object> _args = new List<object>();
+
+        public bool IsWithCursor { get; private set; }
+
+        public ReadOnlyCollection<object> Args => new ReadOnlyCollection<object>(_args);
+
+        public string ArgsString => string.Join(" ", _args.ToArray());
```

Comment:
is this `ToArray` needed? Most TFMs should be fine with just `string.Join(" ", _args);` - also; this isn't "free" - so having it as a property is misleading; suggest `GetArgsString()` unless there's a reason not to?

Reply:
actually, what is the use-case of this API?

Reply:
The only use case that I can see is internal for the tests present in the QueryBuilder\BuilderTest.cs file.

How about I convert to a method and make it internal?